mm2src/lp_ordermatch.rs

```diff
@@ -3285,6 +3361,142 @@ pub async fn set_price(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, Strin
+        match my_maker_orders.get(&req.uuid) {
+            Some(order) => {
+                let base = order.base.as_str();
+                let base_coin: MmCoinEnum = match try_s!(lp_coinfind(&ctx, base).await) {
+                    Some(coin) => coin,
+                    None => return ERR!("Base coin {} has been removed from config", base),
+                };
+
+                let rel = order.rel.as_str();
+                let rel_coin: MmCoinEnum = match try_s!(lp_coinfind(&ctx, rel).await) {
+                    Some(coin) => coin,
+                    None => return ERR!("Rel coin {} has been removed from config", rel),
+                };
+
+                let original_conf_settings = order.conf_settings.unwrap();
+
+                let updated_conf_settings = OrderConfirmationsSettings {
+                    base_confs: req.base_confs.unwrap_or(original_conf_settings.base_confs),
+                    base_nota: req.base_nota.unwrap_or(original_conf_settings.base_nota),
+                    rel_confs: req.rel_confs.unwrap_or(original_conf_settings.rel_confs),
+                    rel_nota: req.rel_nota.unwrap_or(original_conf_settings.rel_nota),
+                };
+
+                (
+                    base_coin,
+                    rel_coin,
+                    order.price.clone(),
+                    order.max_base_vol.clone(),
+                    updated_conf_settings,
+                    order.matches.clone(),
+                )
+            },
+            None => return ERR!("There is no order with UUID {}", req.uuid),
+        };
+
+    drop(my_maker_orders);
+
+    let mut update_msg = new_protocol::MakerOrderUpdated::new(req.uuid);
+    update_msg = update_msg.with_new_conf_settings(updated_conf_settings);
+
+    // Validate and Add new_price to update_msg if new_price is found in the request
+    let new_price = match req.new_price {
+        Some(new_price) => {
+            try_s!(validate_price(new_price.clone()));
+            update_msg = update_msg.with_new_price(new_price.clone().into());
+            new_price
+        },
+        None => original_price,
+    };
+
+    let min_base_amount = base_coin.min_trading_vol();
+    let min_rel_amount = rel_coin.min_trading_vol();
+
+    // Add min_volume to update_msg if min_volume is found in the request
+    if let Some(min_volume) = req.min_volume.clone() {
+        // Validate and Ca lculate Minimum Volume
+        let actual_min_vol = try_s!(validate_and_get_min_vol(
+            min_base_amount.clone(),
+            min_rel_amount.clone(),
+            Some(min_volume),
+            new_price.clone()
+        ));
+        update_msg = update_msg.with_new_min_volume(actual_min_vol.into());
+    }
+
+    // Calculate order volume and add to update_msg if new_volume is found in the request
+    let new_volume = if req.max.unwrap_or(false) {
+        let max_volume = try_s!(get_max_volume(&ctx, &base_coin, &rel_coin).await);
```

Comment:
I think that for the `max` case it might be more intuitive to have `max_volume = get_max_volume + reserved_amount`. The `max` means that the user prefers to trade the entire available balance. Current implementation won't match this desire in the following case:

1. Let's say we have an order with `max_base_vol = 3`.
2. It is partially matched reserving 1, `available` becomes 2.
3. User top-ups the base coin's balance to e.g. 5.
4. Calls `update` with `max:true`.
5. Update will set the `max_base_vol` to 5, but available will be 4 as there is the match reserving 1.

Reply:
But I have just noticed a possible problem - `reserved` can change during balance checks, etc. Might be harder to implement it this way as I initially thought :slightly_smiling_face:

Reply:
Oh, if new matches occur we will return an error, please discard the above comment :slightly_smiling_face:

Reply:
I think it should be `max_volume = get_max_volume +  get_locked_amount(ctx, coin)` since `get_max_volume` subtracts the volume locked in swaps not matches unless I am mistaken ofcourse :)

Reply:
`get_locked_amount` returns the balance that is going to be used in running swaps but not used yet (the transactions are not sent). We shouldn't add it to `max_volume` as it will result in more volume being set than actually available. Example (not taking transaction fees into account for simplicity):

1. Current balance - 5, order A max_base_vol - 3, reserved in order - 1, locked by swaps - 2 (e.g. 2 swaps started with 1 volume - by the current order A and some another order B).
2. `update_maker_order` with `max:true` is called for A, `get_max_volume` returns 3.
3. If we add `get_locked_amount` to it we set the `max_base_vol` for A to 5.
4. Available of A becomes 4, but when swap transactions are sent the balance will be 3.